ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala

```diff
@@ -50,19 +51,20 @@ object SandboxServer {
   private val engine = Engine()
 
   // if requested, initialize the ledger state with the given scenario
-  private def createInitialState(config: SandboxConfig, context: SandboxContext)
+  private def createInitialState(config: SandboxConfig, packages: SandboxPackageStore)
     : (ActiveContractsInMemory, ImmArray[LedgerEntryWithLedgerEndIncrement], Option[Instant]) = {
     // [[ScenarioLoader]] needs all the packages to be already compiled --
     // make sure that that's the case
     if (config.eagerPackageLoading || config.scenario.nonEmpty) {
-      for ((pkgId, pkg) <- context.packageContainer.packages) {
+      for (pkgId <- packages.listLfPackagesSync().keys) {
```

Comment:
Currently, the whole sandbox initialization part is hardcoded to work with in-memory data structures:
- The scenario loader depends on `ActiveContractsInMemory` and `InMemoryPackageStore`
- The scenario loader uses `listLfPackagesSync`, and does not use any async code.
- The scenario loader uses its own ledger implementation

The scenario is executed before the postgres ledger is initialized. It would be a bigger change to refactor the scenario to be executed on the postgres ledger.

Reply:
I see. Makes sense then.